### DISCUSSION TITLE
Using RAG though the API

### DISCUSSION THREAD

**Discussion by energy29uk** [open-webui/open-webui#3804] [General]
Hi, is it possible to use the RAG functionality through the API and if so does anyone have an example  ? I've searched around for weeks and can perform the basic "Why is the sky blue" call but that's about it.

**Comment by crawfordg** [comment]
Get anywhere on this? Having same issue

**Reply by energy29uk** [reply]
No nothing as yet. I did try embedding some documents into a custom model,  but it doesn't seem to push that back to Ollama so the API returned something like "Model Not found". It's a great project but with so little supporting documentation I've had to start looking to see if anything will do it.

**Comment by PlebeiusGaragicus** [comment]
This would be incredible!  I want OI to expose a vector database that I can query how I need to in my own agent flows.  It's so much more powerful than the current implementation.
